pkg/agent/sysadvisor/plugin/qosaware/resource/cpu/region/region_dedicated_numa_exclusive.go

```diff
@@ -86,3 +81,54 @@ func (r *QoSRegionDedicatedNumaExclusive) TryUpdateProvision() {
 		internal.updateStatus = types.PolicyUpdateSucceeded
 	}
 }
+
+func (r *QoSRegionDedicatedNumaExclusive) getControlKnobs() types.ControlKnob {
+	reclaimedCPUSize := 0
+	if reclaimedInfo, ok := r.metaReader.GetPoolInfo(state.PoolNameReclaim); ok {
+		for _, numaID := range r.bindingNumas.ToSliceInt() {
+			reclaimedCPUSize += reclaimedInfo.TopologyAwareAssignments[numaID].Size()
+		}
+	}
+	cpuRequirement := r.ResourceUpperBound + r.ReservedForReclaim - float64(reclaimedCPUSize)
+
+	return types.ControlKnob{
+		types.ControlKnobNonReclaimedCPUSize: {
+			Value:  cpuRequirement,
+			Action: types.ControlKnobActionNone,
+		},
+	}
+}
+
+func (r *QoSRegionDedicatedNumaExclusive) getIndicators() types.Indicator {
+	var (
+		cpi float64
+		err error
+	)
+
+	// get cpi of the main container in this region
+	for podUID, containerSet := range r.podSet {
+		for containerName := range containerSet {
+			ci, ok := r.metaReader.GetContainerInfo(podUID, containerName)
+			if !ok || ci == nil {
+				klog.Errorf("[qosaware-cpu] illegal container info of %v/%v", podUID, containerName)
+				return nil
+			}
+			if ci.ContainerType == v1alpha1.ContainerType_MAIN {
```

Comment:
this coding style is kind of wired: it assumes that that there should be only one valid pod/container (the latter cpi will always override the previous one); but there is nowhere to limit this, so, i'd rather to sum up the cpi for all those pod/containers in this region